chkphy

```diff
@@ -19,19 +19,26 @@
 
 
 # first we sleep.
-# this gives plently of time for linux to mess with the RTC 
-# (where does it do that set?) before we touch it. Otherwise it 
+# this gives plently of time for linux to mess with the RTC
+# (where does it do that set?) before we touch it. Otherwise it
 # occasionally updates it right between when we are doing our sets.
 
 # This also gives you a chance to abort in case something goes wrong
 # and avoid a infinite loop
 
+kernel_major=`uname -r | grep -o -E '^[0-9]+'`
+
 sleep 30
 
-if [[ $(phyreg test 18 13) == "1" ]]; then 
+if [[ $(phyreg test 18 13) == "1" ]]; then
 
 	sleep 1
 
+  # unlock rtc registers if we are running a kernel newer than 3.X
+  if [ "$kernel_major" -ge "4" ]; then
```

Comment:
Just curious: is there any downside to unlocking these if they are already unlocked?  If not, could avoid the version check and just always unlock?